imports/methods/announcements/post.ts

```diff
@@ -0,0 +1,38 @@
+import { check } from 'meteor/check';
+import { ValidatedMethod } from 'meteor/mdg:validated-method';
+import { Meteor } from 'meteor/meteor';
+import Ansible from '../../Ansible';
+import Announcements from '../../lib/models/Announcements';
+import MeteorUsers from '../../lib/models/MeteorUsers';
+import PendingAnnouncements from '../../lib/models/PendingAnnouncements';
+import { userMayAddAnnouncementToHunt } from '../../lib/permission_stubs';
+
+export default new ValidatedMethod({
+  name: 'Announcements.methods.post',
+
+  validate({ huntId, message }: { huntId: unknown, message: unknown }) {
+    check(this.userId, String);
+    check(huntId, String);
+    check(message, String);
+
+    if (!userMayAddAnnouncementToHunt(this.userId, huntId)) {
+      throw new Meteor.Error(401, `User ${this.userId} may not create announcements for hunt ${huntId}`);
+    }
+  },
+
+  run({ huntId, message }: { huntId: string, message: string }) {
```

Comment:
I don't love that we write the method signature twice here, once for validation purposes, and once for "actually do the thing" purposes, and (because of the limitations in the approach) they actually declare different types and there's no validation that the result of `validate()` is what we're passing on to `run()`.  It's creating a new possible class of "these things do not match" errors that is likely to escape detection by our tooling, which doesn't feel like much of a step forward to me.